mod/calculated_metrics/set/type/metric/dependencies.rb

```diff
@@ -10,18 +10,22 @@ def each_dependent_metric
   dependency_tree.each_metric { |m| yield m }
 end
 
+# note: #formula_metrics will find score metrics when scored by formula
+# but not when scored by mapping.
 def directly_dependent_metrics
-  score_metrics + formula_metrics
+  (score_metrics + formula_metrics).uniq
 end
 
 def dependency_tree
   DependencyTree.new directly_dependent_metrics
 end
 
 def score_metrics
-  Card.search type_id: MetricID, left_id: id
+  @score_metrics ||=
+    Card.search type_id: MetricID, left_id: id
```

Comment:
Are you sure that this does no harm?  I've seen too often how this kind of caching blew up something

Reply:
`Sure` would be an overstatement, but the only place this is called is in `directly_dependent_metrics`, which also calls `formula_metrics`, which already uses this kind of caching.  It seemed odd to me to use it in the one but not the other. I've also tried (and failed) to concoct a scenario when it would be a problem. But if those don't seem like reason enough, I can undo it.

Reply:
verdict?

Reply:
acceptable